structs with flexible array member; that might be not safe

For example, `tube` has `struct job buried` as last member, while `struct job` has `tube tube` embed in the middle. `struct jobs buried` means that this struct is embed, while `tube tube` means the pointer to the tube struct. This produces warnings if I try to compiler with `-pedantic`. Also it is hard to look and reason about this because I have to constantly think that `tube` is in fact a `*tube` (see #458).

As first step we need to fix #458 and maybe compiler will not warn anymore. If not, then we might need to rearrange the structures, however, this is not trivial. 

```
~/d/beanstalkd (master=) % make clean all CFLAGS="-std=gnu99 -Wextra -Wflexible-array-extensions"
rm -f vers.c  beanstalkd primes.o sd-daemon.o vers.o
cc -std=gnu99 -Wextra -Wflexible-array-extensions -Wall -Werror -Wformat=2 -g   -c -o darwin.o darwin.c
In file included from darwin.c:1:
./dat.h:205:16: error: 'buried' may not be nested in a struct due to flexible array member [-Werror,-Wflexible-array-extensions]
    struct job buried;
               ^
./dat.h:336:16: error: 'reserved_jobs' may not be nested in a struct due to flexible array member [-Werror,-Wflexible-array-extensions]
    struct job reserved_jobs; // linked list header
               ^
./dat.h:396:16: error: 'jlist' may not be nested in a struct due to flexible array member [-Werror,-Wflexible-array-extensions]
    struct job jlist; // jobs written in this file
               ^
3 errors generated.
make: *** [darwin.o] Error 1
```

--- a/dat.h
+++ b/dat.h
@@ -213,7 +213,7 @@ struct Job {
     int walresv;
     int walused;
 
-    char body[];                // written separately to the wal
+    char *body;                 // written separately to the wal
 };
 
 struct Tube {
@@ -221,13 +221,13 @@ struct Tube {
     char name[MAX_TUBE_NAME_LEN];
     Heap ready;
     Heap delay;
-    Ms waiting; /* set of conns */
+    Ms waiting;                 // set of conns
     struct stats stat;
     uint using_ct;
     uint watching_ct;
     int64 pause;
     int64 deadline_at;
-    Job buried;
+    Job buried;                 // linked list header
 };
 
 

--- a/job.c
+++ b/job.c
@@ -95,12 +95,16 @@ allocate_job(int body_size)
     Job *j;
 
     j = malloc(sizeof(Job) + body_size);
-    if (!j) return twarnx("OOM"), (Job *) 0;
+    if (!j) {
+        twarnx("OOM");
+        return (Job *) 0;
+    }
 
     memset(j, 0, sizeof(Job));
     j->r.created_at = nanoseconds();
     j->r.body_size = body_size;
-    j->next = j->prev = j; /* not in a linked list */
+    j->body = (char *)j + sizeof(Job);
+    j->next = j->prev = j;      // not in a linked list
     return j;
 }
 

